### DISCUSSION TITLE
Specify a base page and a destination on the event opening page to update the specified subpages when using a mixed container

### DISCUSSION THREAD

**Discussion by fanhai** [frangoteam/FUXA#925] [Ideas]
Specify a base page and a destination on the event opening page to update the specified subpages when using a mixed container
在事件打开页面上指定一个基础页面和一个目的地，以便在使用混合容器时更新指定子页面

First, divide the mixed container into several sub pages, each of which serves as a base page. You can place the base control page or drag and drop other pages to combine.
先将混合容器分成几个子页面，每个子页面作为一个基础页面，可以放基础控件页可以拖放其他页面进行组合。
如果这个页面直接拖放切换页面的按钮，按钮的Events可以指定基础子页面

If you drag and drop the switch button directly on this page, the Events of the button can specify the basic sub page.
如果这个页面直接拖放切换页面的按钮，按钮的Events可以指定基础子页面

**Comment by unocelli** [comment]
Hi, interesting...I would be grateful if you could make me a small drawing

**Reply by fanhai** [reply]
Something went really wrong, and we can’t process that file.  I cannot upload images

**Reply by fanhai** [reply]
Like this, click on button1 button2 button3 to switch to subview1 and subview1 respectively_ 2 subview1_ 3, as shown in the following figure

**Comment by fanhai** [comment]
![image](https://github.com/frangoteam/FUXA/assets/9838830/086473b0-19a8-429d-a794-f7597490b283)
像这样，点击button1  button2 button3 分别切换到 subview1 subview1_2 subview1_3

**Reply by fanhai** [reply]
![image](https://github.com/frangoteam/FUXA/assets/9838830/1e14e8a7-1981-498f-96f8-2887e130a2b2)

**Comment by RobertBudgen** [comment]
I am also looking for similar but for a device dialog.
I would like to display a historical graph on my popup.
The Dialog Tags are bind to the PLC tags when the user clicks on the device.
Like take a PID loop for Example 3 Main Tags, PV,CV & SP
the  3 key tags are data logged, PLC tags PID[0].PV, PID[1].PV etc.... 
Local Fuxa tags PV,CV &  SP are bind to the associated PLC TAG Local PV Bind to plc.PID[0].PV etc...

Now the graph issue.
My thoughts were to utilize a MVC with both the Dialog and a graph component.
the dialog part would always use the bind tags but the graph component would need to be a page for PID[0], PID[1], PID[2] displaying the correct graph. When user clicks the PID the page opens with the configuration dialog all binded to the PLC tags and the historical graph

I did try to create some local tags Graph_PV, Graph_CV & Graph_SP and created a graph containing the 3  local tags. on the  hope that on bind the  tag UUID was passed and the correct data from the  data base would be displayed. Nice Idea but no Joy.

**Reply by unocelli** [reply]
@RobertBudgen Hi, I suppose that is the same missing feature #859

**Comment by RobertBudgen** [comment]
Similar. I went are all looking for away to A, specify a chart or chart variables on opening a screen.  Parameters or script. B, control the contents of MVC fromEither script or parameters In originally took the MVC as a single page made up from other pages as Objects / Tiles. Not a multi page view. With each view operating independently. A common feature would be like a motor graphic. The same page dropped on a MVC in multiple places with multiple configurations Connections would be binded to the each MVC. Basically graphical objects. This is quite a lot of work and probably not practical but the popup binding works well apart from if you wish to show a trend  Best regards,Robert BudgenSystem Engineer Technical Solutions ExpertBürkert Fluid Control SystemsBurkert Contromatic Inc.5002 South Service RoadBurlington ON  L7L 5Y7Cell:  (905) 630-0694Phone: (905) 632-3033Fax:     (905) ***@***.*** from IPhoneOn Nov 10, 2023, at 4:31 PM, Umberto Nocelli ***@***.***> wrote:﻿
@RobertBudgen Hi, I suppose that is the same missing feature #859

**Reply by unocelli** [reply]
Hi, sorry I'm late, I'm not sure I understand what you mean, maybe a small sketch would help

**Comment by robbudge** [comment]
The Same Page could be used but configurable in the MVC or other page
![image](https://github.com/frangoteam/FUXA/assets/15365138/2e566550-75f0-46af-8d43-53fc655dcd61)
or
operates as a single page, clicks on page  1, displays either Page 1a or Page 1b
![image](https://github.com/frangoteam/FUXA/assets/15365138/3c8dd970-deb5-41e6-9f32-accfd73fb73e)

Currently correct me if i'm wrong the MVC operates almost like Split-Screen. Allowing Multiple pages  to fully function independently within the same window.

**Reply by fanhai** [reply]
How to configure it？

**Reply by robbudge** [reply]
Currently it’s a request. 
 my thoughts would be when you create the MVC container you specify the local / virtual tags, like server tags. 
 
Imagine each container being a virtual device in the connection manager. 

Each container / Virtual device would have its own tags. 

When showing the container you could bind the device tags from the physical to the virtual. 

So one option utilizing grouping. 
In connections I create a Virtual Device called motor with the tags foo & bar 

I create special page with a group called motor, utilizing the virtual tags  foo & barr within the graphics. 

Each time I deploy the group motor, Fuxa Creates a Virtual device (‘Motor_1’ group name )with the tags foo & barr that could be binded. 

any changes to the primary group would be reflected in the child deployments. 

under properties of the child you would have the binding option. 


I think tag binding via script would a great step in this direction. 
dialogs work well with the binding but if you have a dialog with 20 tags it can be tedious. 
a script option would be great especially if your utilizing arrays in the PLC. 

Passing the prefix to a ‘Motor Bind’ script. 
where binding is done via script. 

Ie
Server.motor_current := Bind(prefix+’.motor_current’)
Server.hz:=bind(prefix+’.hz’)

The bind would return the UUID of the passed string.

**Reply by henjoe** [reply]
+1 for this feature. 
Basically you can create your templated Graphics (could be a View) with a corresponding vietual tags that can be bind to physical tags, or you can also create a Graphic component (motor graphics, faceplate and etc popup) and add some virtual tags on it that can be also bind to physical tag. You can use them anywhere and they have separate instances of each other.

**Comment by unocelli** [comment]
Hi,
I have just added the Panel control which should be of relevance to this discussion and your feedback would be interesting

**Reply by robbudge** [reply]
Ok, Thanks,i will pull the  files and test today.

**Reply by robbudge** [reply]
How is the Panel 'Tag' utilized within the selected view.
The actual Panel Block works great, if i  could pass either a Tag Prefix or multiple tags

**Reply by unocelli** [reply]
Hi, we are working to a new feature 'placeholder' that should open more flexibility to best resolve these issues

**Comment by robbudge** [comment]
It works really well and now allows me to specify a Graph page when opening a popup. 

The part I was missing was. 
declare a server string tag ‘panel name’
Create multiple pages each with the required graphics. 
panel_1, panel_2, panel_3 all the same size but with different graphs. 

Add the panel item to the main page set the variable to ‘panel name’

Then when I open a dialog I set ‘panel_name’ to the relevant panel page 
Basically set panel_name to panel_1
And the correct sub page is shown in my dialog with the existing controls. 

Now with alias tags and ‘panel_name’ I can show not only the control of a PID loop but a historical trend.
